syz-manager/manager.go

```diff
@@ -532,9 +532,18 @@ func (mgr *Manager) loadProg(data []byte, minimized, smashed bool) bool {
 	}
 	if disabled {
 		// This program contains a disabled syscall.
-		// We won't execute it, but remember its hash so
-		// it is not deleted during minimization.
+		// Remember its hash so that it remains in the corpus.
 		mgr.disabledHashes[hash.String(data)] = struct{}{}
+		leftover := programLeftover(mgr.target, mgr.targetEnabledSyscalls, data)
```

Comment:
Maybe it should happen only if !PreserveProgs. Otherwise if I have a large corpus and enable just few syscalls (which I frequently for for a single subsystem testing), I will get thousands of programs for triage each time.
Doing this when !PreserveProgs makes sense logically: programs are either preserved as-is for future, or they are given a second chance before being discarded.

Reply:
Makes sense. Updated the PR.